src/ucp/proto/proto_am.inl

```diff
@@ -522,4 +539,61 @@ ucp_proto_ssend_ack_request_alloc(ucp_worker_h worker, ucs_ptr_map_key_t ep_id)
     return req;
 }
 
+static UCS_F_ALWAYS_INLINE ucs_status_t
+ucp_am_bcopy_handle_status_from_pending(uct_pending_req_t *self, int multi,
+                                        ucs_status_t status)
+{
+    ucp_request_t *req;
+
+    if (multi) {
+        if (status == UCS_INPROGRESS) {
+            return UCS_INPROGRESS;
+        }
+
+        if (ucs_unlikely(status == UCP_STATUS_PENDING_SWITCH)) {
+            return UCS_OK;
+        }
+    } else {
+        ucs_assert(status != UCS_INPROGRESS);
+    }
+
+    if (ucs_unlikely(status == UCS_ERR_NO_RESOURCE)) {
+        return UCS_ERR_NO_RESOURCE;
+    }
+
+    req = ucs_container_of(self, ucp_request_t, send.uct);
+    ucp_request_send_generic_dt_finish(req);
+    ucp_request_complete_send(req, status);
+    return UCS_OK;
+}
+
+static UCS_F_ALWAYS_INLINE ucs_status_t
+ucp_am_bcopy_sync_handle_status_from_pending(uct_pending_req_t *self, int multi,
```

Comment:
maybe unite with previous func and add another parameter (compile time constant) indicating whether it is sync send or not